From e9ce9dc2d87372c937bb4ce41b3a69798646a948 Mon Sep 17 00:00:00 2001
From: Byron McCollum <byron.mccollum@rackspace.com>
Date: Fri, 12 Feb 2016 23:49:10 -0600
Subject: [PATCH] [WIP] Accommodate Rabbitless Stand-Alone Swift Installations

When RabbitMQ is not installed (affinity: 0), because it is not used
in a Stand-Alone Swift installation, do not setup messaging vhost and
user for the various services.

Note: The only change that is actually needed is in Keystone, but all
other services have been updated for consistency and clarity.

Change-Id: I3016039692d8130654fe1bff422f24ef2afc196e
---
 playbooks/roles/os_aodh/tasks/aodh_pre_install.yml             | 3 ++-
 playbooks/roles/os_ceilometer/tasks/ceilometer_pre_install.yml | 3 ++-
 playbooks/roles/os_cinder/tasks/cinder_pre_install.yml         | 3 ++-
 playbooks/roles/os_glance/tasks/glance_pre_install.yml         | 3 ++-
 playbooks/roles/os_heat/tasks/heat_pre_install.yml             | 3 ++-
 playbooks/roles/os_keystone/tasks/keystone_pre_install.yml     | 3 ++-
 playbooks/roles/os_neutron/tasks/neutron_pre_install.yml       | 3 ++-
 playbooks/roles/os_nova/tasks/nova_pre_install.yml             | 3 ++-
 8 files changed, 16 insertions(+), 8 deletions(-)

diff --git a/playbooks/roles/os_aodh/tasks/aodh_pre_install.yml b/playbooks/roles/os_aodh/tasks/aodh_pre_install.yml
index 6e4a820..95efaff 100644
--- a/playbooks/roles/os_aodh/tasks/aodh_pre_install.yml
+++ b/playbooks/roles/os_aodh/tasks/aodh_pre_install.yml
@@ -90,4 +90,5 @@
 
 - include: aodh_messaging_setup.yml
   when: >
-    inventory_hostname == groups['aodh_api'][0]
+    inventory_hostname == groups['aodh_api'][0] and
+    groups['rabbitmq_all']|length > 0
diff --git a/playbooks/roles/os_ceilometer/tasks/ceilometer_pre_install.yml b/playbooks/roles/os_ceilometer/tasks/ceilometer_pre_install.yml
index bec4f32..ca38403 100644
--- a/playbooks/roles/os_ceilometer/tasks/ceilometer_pre_install.yml
+++ b/playbooks/roles/os_ceilometer/tasks/ceilometer_pre_install.yml
@@ -105,4 +105,5 @@
 
 - include: ceilometer_messaging_setup.yml
   when: >
-    inventory_hostname == groups['ceilometer_api'][0]
+    inventory_hostname == groups['ceilometer_api'][0] and
+    groups['rabbitmq_all']|length > 0
diff --git a/playbooks/roles/os_cinder/tasks/cinder_pre_install.yml b/playbooks/roles/os_cinder/tasks/cinder_pre_install.yml
index 0c65a5d..44fd191 100644
--- a/playbooks/roles/os_cinder/tasks/cinder_pre_install.yml
+++ b/playbooks/roles/os_cinder/tasks/cinder_pre_install.yml
@@ -102,4 +102,5 @@
 
 - include: cinder_messaging_setup.yml
   when: >
-    inventory_hostname == groups['cinder_all'][0]
+    inventory_hostname == groups['cinder_all'][0] and
+    groups['rabbitmq_all']|length > 0
diff --git a/playbooks/roles/os_glance/tasks/glance_pre_install.yml b/playbooks/roles/os_glance/tasks/glance_pre_install.yml
index 4686413..d97000a 100644
--- a/playbooks/roles/os_glance/tasks/glance_pre_install.yml
+++ b/playbooks/roles/os_glance/tasks/glance_pre_install.yml
@@ -95,4 +95,5 @@
 
 - include: glance_messaging_setup.yml
   when: >
-    inventory_hostname == groups['glance_all'][0]
+    inventory_hostname == groups['glance_all'][0] and
+    groups['rabbitmq_all']|length > 0
diff --git a/playbooks/roles/os_heat/tasks/heat_pre_install.yml b/playbooks/roles/os_heat/tasks/heat_pre_install.yml
index 7fc1ec9..be7c1c7 100644
--- a/playbooks/roles/os_heat/tasks/heat_pre_install.yml
+++ b/playbooks/roles/os_heat/tasks/heat_pre_install.yml
@@ -103,4 +103,5 @@
 
 - include: heat_messaging_setup.yml
   when: >
-    inventory_hostname == groups['heat_all'][0]
+    inventory_hostname == groups['heat_all'][0] and
+    groups['rabbitmq_all']|length > 0
diff --git a/playbooks/roles/os_keystone/tasks/keystone_pre_install.yml b/playbooks/roles/os_keystone/tasks/keystone_pre_install.yml
index 3fb9eb1..75b737c 100644
--- a/playbooks/roles/os_keystone/tasks/keystone_pre_install.yml
+++ b/playbooks/roles/os_keystone/tasks/keystone_pre_install.yml
@@ -121,4 +121,5 @@
 
 - include: keystone_messaging_setup.yml
   when: >
-    inventory_hostname == groups['keystone_all'][0]
+    inventory_hostname == groups['keystone_all'][0] and
+    groups['rabbitmq_all']|length > 0
diff --git a/playbooks/roles/os_neutron/tasks/neutron_pre_install.yml b/playbooks/roles/os_neutron/tasks/neutron_pre_install.yml
index cea44b0..fc1bb0d 100644
--- a/playbooks/roles/os_neutron/tasks/neutron_pre_install.yml
+++ b/playbooks/roles/os_neutron/tasks/neutron_pre_install.yml
@@ -107,4 +107,5 @@
 
 - include: neutron_messaging_setup.yml
   when: >
-    inventory_hostname == groups['neutron_all'][0]
+    inventory_hostname == groups['neutron_all'][0] and
+    groups['rabbitmq_all']|length > 0
diff --git a/playbooks/roles/os_nova/tasks/nova_pre_install.yml b/playbooks/roles/os_nova/tasks/nova_pre_install.yml
index 8f66a0d..e2cac6e 100644
--- a/playbooks/roles/os_nova/tasks/nova_pre_install.yml
+++ b/playbooks/roles/os_nova/tasks/nova_pre_install.yml
@@ -124,4 +124,5 @@
 
 - include: nova_messaging_setup.yml
   when: >
-    inventory_hostname == groups['nova_all'][0]
+    inventory_hostname == groups['nova_all'][0] and
+    groups['rabbitmq_all']|length > 0
-- 
2.7.1

